Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2024 Colin Jones. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Colin Jones
+-/
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Tactic.NormNum.Prime
+import Mathlib.Algebra.GeomSum
+import Mathlib.Tactic.FinCases
+
+/-!
+# Factorisation properties of natural numbers
+This file defines abundant, pseudoperfect, deficient, and weird numbers and formalizes their
+  relations with prime and perfect numbers.
+
+## Main Definitions
+* `Nat.Abundant`: a natural number `n` is _abundant_ if the sum of its proper divisors are greater
+  than `n`
+* `Nat.Pseudoperfect`: a natural number `n` is _pseudoperfect_ if a subset of its proper divisors
+  equals `n`
+* `Nat.Deficient`: a natural number `n` is _deficient_ if the sum of its proper divisors is less
+  than `n`
+* `Nat.Weird`: a natural number is _weird_ if it is both abundant and not pseudoperfect
+
+## Main Results
+* `Nat.deficient_or_perfect_or_abundant`: A positive natural number is either deficient,
+  perfect, or abundant.
+* `Nat.prime_deficient`: All prime natural numbers are deficient.
+* `Nat.exists_infinite_deficients`: There are infinitely many deficient numbers.
+* `Nat.prime_pow_deficient`: Any natural number power of a prime is deficient.
+
+## Implementation Notes
+* Zero is not included in any of the definitions and these definitions only apply to natural
+  numbers greater than zero.
+
+## References
+* Prielipp, Robert W. “PERFECT NUMBERS, ABUNDANT NUMBERS, AND DEFICIENT NUMBERS.”
+  The Mathematics Teacher, vol. 63, no. 8, 1970, pp. 692–96. JSTOR,
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+
+abundant, deficient, weird, pseudoperfect
+-/
+
+open BigOperators Finset
+
+namespace Nat
+
+variable {n m p : ℕ}
+
+/-- `n : ℕ` is _abundant_ if the sum of the proper divisors of `n` is greater than `n`. -/
+def Abundant (n : ℕ) : Prop := ∑ i in properDivisors n, i > n
+
+/-- `n : ℕ` is _deficient_ if the sum of the proper divisors of `n` is less than `n`. -/
+def Deficient (n : ℕ) : Prop := ∑ i in properDivisors n, i < n
+
+/-- A positive natural number `n` is _pseudoperfect_ if there exists a subset of the proper
+  divisors of `n` such that the sum of that subset is equal to `n`. -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s ⊆ properDivisors n, ∑ i ∈ s, i = n
+
+/-- `n : ℕ` is a _weird_ number if and only if it is both abundant and pseudoperfect. -/
```

Comment:
```suggestion
/-- `n : ℕ` is a _weird_ number if and only if it is abundant but not pseudoperfect. -/
```

Reply:
I don't think this suggestion was applied - the docstring doesn't seem to match the definition to me